Fix: RuntimeError: CRC Mismatch / ATECC608A not reachable via I2C anymore

rrottmann opened the pull request:
- Fix for adafruit/Adafruit_CircuitPython_ATECC#16
- According to ATECC608A datasheet, the config byte 16 is indeed wrong and sets I2C address 0x10
- Personalization of ATECC608 not tested as I have no unprovisioned/spare modules left
- Added documentation for i2c config byte 16
- Added I2C address variable used to derive config byte 16
- Added I2C address to config byte 16 of CFG_TLS
- Modified representation of CFG_TLS

rrottmann committed:
- Added documentation for config bytes
- Added documentation for i2c config byte 16
- Added I2C address variable used to derive config byte 16
- Added I2C address to config byte 16 of CFG_TLS
- Modified representation of CFG_TLS

brentru commented on adafruit_atecc/adafruit_atecc.py:
```diff
@@ -52,8 +52,20 @@
+_I2C_ADDR = 0x60
+_REG_ATECC_ADDR = _convert_i2c_addr_to_atecc_addr(i2c_addr=_I2C_ADDR)
+
+# TODO: Verify that _REG_ATECC_ADDR is still 0xC0
```
are these TODO's supposed to still be here?

rrottmann replied:
I've added them just for the review in order to give some pointers of what to verify.
For example the different representation of the configuration bytes look quite drastic and I wanted to show that I just modify the I2C configuration in byte 16. As the config can only be applied once it should be clear that the changes will do no harm.

brentru replied:
ok!

brentru commented:
Hi, thanks for the PR - does this go with your notes (https://gist.github.com/rrottmann/292d1fb7f30b448e96cb4dedadcaa213)?

I'll review everything once it's passing the CI (https://learn.adafruit.com/improve-your-code-with-pylint).

> Personalization of ATECC608 not tested as I have no unprovisioned/spare modules left

Do you want to add a usage example into `examples/` for this feature?

rrottmann commented:
> Hi, thanks for the PR - does this go with your notes (https://gist.github.com/rrottmann/292d1fb7f30b448e96cb4dedadcaa213)?

I had a look in the datasheet and tried to understand how the I2C is being configured. As I am sure now that this part is set in the config bytes, I made an attempt to fix it.

> I'll review everything once it's passing the CI (https://learn.adafruit.com/improve-your-code-with-pylint).

Will have a look at the code and make pylint happy.

> > Personalization of ATECC608 not tested as I have no unprovisioned/spare modules left
> 
> Do you want to add a usage example into `examples/` for this feature?

With I2C address of 0x60 the behavior should not have changed. During personalization the I2C address is being converted to the proper config byte 16. This overrides the 0x20 found in the original code. That way the personalization should work as intended and the device should be reachable at I2C address 0x60.
When using a locked chip with a "wrong" I2C address, the user may change the variable and still be able to talk with the device.

I try to connect an ATECC608 with wrong I2C address to my dev workstation and will add a small example how to make it work.

rrottmann committed:
- Making pylint happy
- Moved I2C Config doc to CFG_TLS_HEX description
- Corrected BIN representation of 0x6A

rrottmann committed:
- Making pylint happy
- Moved I2C Config doc to CFG_TLS_HEX description
- Corrected BIN representation of 0x6A

rrottmann committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	adafruit_atecc/adafruit_atecc.py

rrottmann committed:
Make pylint happy.

rrottmann committed:
Make black happy.

rrottmann committed:
Fixed typo.

rrottmann committed:
Fixed binary representation.

brentru reviewed (approved):
@rrottmann This all looks good to me - are you still interested in adding an example, or could I merge this PR?

rrottmann commented:
You can merge for now. Will add the example later.
